src/android/BatteryListener.java

```diff
@@ -129,8 +129,18 @@ private void removeBatteryListener() {
     private JSONObject getBatteryInfo(Intent batteryIntent) {
         JSONObject obj = new JSONObject();
         try {
-            obj.put("level", batteryIntent.getIntExtra(android.os.BatteryManager.EXTRA_LEVEL, 0));
-            obj.put("isPlugged", batteryIntent.getIntExtra(android.os.BatteryManager.EXTRA_PLUGGED, -1) > 0 ? true : false);
+            boolean batteryPresent = batteryIntent.getBooleanExtra(android.os.BatteryManager.EXTRA_PRESENT, true);
+            int levelScale = batteryIntent.getIntExtra(android.os.BatteryManager.EXTRA_SCALE, 100);
+
+            // Work around the fact that Pegasus doesn't report cable plugged state correctly.
+            obj.put("isPlugged", batteryPresent
+                    ? (batteryIntent.getIntExtra(android.os.BatteryManager.EXTRA_PLUGGED, -1) > 0 ? true : false)
+                    : true);  // If there's no battery, then we *must* be plugged in.
+
+            obj.put("level", batteryPresent
+                    ? ((batteryIntent.getIntExtra(android.os.BatteryManager.EXTRA_LEVEL, 0) * 100) / levelScale)
+                    : levelScale);  // If there's no battery, lie to cts/mtab and say it's fully charged
```

Comment:
instead of having multi-level ternaries inside of the `obj.put`, I'd say split it out like:

```
boolean batteryPresent = blah;
int levelScale = blah;
boolean isPlugged = batteryIntent.getIntExtra(android.os.BatteryManager.EXTRA_PLUGGED, -1) > 0 ? true : false;
int level = batteryIntent.getIntExtra(android.os.BatteryManager.EXTRA_LEVEL, 0) * 100) / levelScale;

if (!isPresent) {
  isPlugged = true;
  // If there's no battery, lie to cts/mtab and say it's fully charged
  level = levelScale;
}

obj.put("isPlugged", isPlugged);
obj.put("level", level);
```

Reply:
Sure.